packages/api/src/modules/external-integrations/guards/external-integrations-auth.guard.ts

```diff
@@ -0,0 +1,36 @@
+import {
+  BadRequestException,
+  CanActivate,
+  ExecutionContext,
+  Injectable,
+  UnauthorizedException,
+} from '@nestjs/common';
+
+import { UsersService } from '../../users/users.service';
+
+@Injectable()
+export class ExternalIntegrationsAuthGuard implements CanActivate {
+  constructor(private readonly usersService: UsersService) {}
+
+  public async canActivate(context: ExecutionContext): Promise<boolean> {
+    const request = context.switchToHttp().getRequest<{ headers: ExternalIntegrationsHeaders }>();
+    const { authorization, from } = request.headers;
+    const trimmedToken = authorization?.trim();
+
+    if (!trimmedToken || !from) {
+      throw new BadRequestException();
+    }
+
+    try {
+      const foundUser = await this.usersService.findOneByEmail(from);
```

Comment:
Shouldnt you take email from token instead of from headers where anyone can put anything?